### ISSUE TITLE
rolldown panics when bundling express

### ISSUE THREAD

**Issue by camc314** [rolldown/rolldown#1443]
```
[crates/rolldown/src/module_finalizers/scope_hoisting/impl_visit_mut.rs:366:9] &self.ctx.module.debug_resource_id = "node_modules/express/lib/view.js"
[crates/rolldown/src/module_finalizers/scope_hoisting/impl_visit_mut.rs:367:9] &call_expr.span = Span {
    start: 1550,
    end: 1562,
}
thread '<unnamed>' panicked at crates/rolldown/src/module_finalizers/scope_hoisting/impl_visit_mut.rs:368
no entry found for key
```

offending express code:

```ts
  if (!opts.engines[this.ext]) {
    // load engine
    var mod = this.ext.slice(1)
    debug('require "%s"', mod)

    // default engine export
    var fn = require(mod).__express

    if (typeof fn !== 'function') {
      throw new Error('Module "' + mod + '" does not provide a view engine.')
    }

    opts.engines[this.ext] = fn
  }
```

(the issue is caused by `require`)

**Comment by camc314** [comment]
related: 
 - https://github.com/expressjs/express/issues/4503
 - https://stackoverflow.com/questions/65009722/bundling-expressjs-with-esbuild-gives-missing-view-js-warning
 - https://github.com/expressjs/express/issues/4491
 - https://corpglory.com/s/express-webpack/

**Comment by hyf0** [comment]
Thanks for reporting. Some edge cases aren't handled well. Just fixed this in #1445.

**Comment by camc314** [comment]
> Thanks for reporting. Some edge cases aren't handled well. Just fixed this in #1445.

thank you.
